test/getApp.test.js

```diff
@@ -0,0 +1,71 @@
+const {getBearerToken} = require('../src/app/auth')
+const faker = require('faker')
+const sinon = require("sinon")
+const axios = require('axios')
+
+let req
+let resSpy = {}
+let authToken
+let nextSpy
+let serverUrl = 'http://localhost:3000/app'
+let headers
+
+
+describe('Retrieve app details - Integration Tests', () => {
+	beforeAll(() => {
+		req = {method: 'GET', query: {}, headers: {}}
+		resSpy.send = sinon.spy()
+		resSpy.status = sinon.spy()
+		nextSpy = sinon.spy()
+		headers = {}
+	})
+
+	describe('Get App details authorization tests', () => {
+		it('Should fail Missing authorization in header', async () => {
+			expect.assertions(1)
+			await expect(axios.get(`${serverUrl}/com.instagram.android`,
+				{headers: headers})).rejects.toThrow('Request failed with status code 401')
+		})
+
+		it('Should fail dude to unauthorized token', async () => {
+			headers['authorization'] = 'eyJhbGciOiJIUzI1NiJ9.eyJ1c2VyaWQiOiJqb2huLmRvZSJ9.CuScq77_iCP4XsYGCMgGnQiATOmQwu_rR1LEB2Pcd_I'
+			expect.assertions(1)
+			await expect(axios.get(`${serverUrl}/com.instagram.android`,
+				{headers: headers})).rejects.toThrow('Request failed with status code 401')
+		})
+
+		it('Should pass with Authorized User', async () => {
+			req['query']['userName'] = faker.name.findName()
+			req['query']['password'] = faker.internet.password()
+			await getBearerToken(req, resSpy)
+			authToken = resSpy.send.args[0][0]
+			headers['authorization'] = authToken
+			const response = await axios.get(`${serverUrl}/com.strava`, {headers: headers})
+			expect(response).toMatchObject({status: 200})
+			expect(response).toMatchObject({statusText: 'OK'})
+			expect(response.data).toMatchObject({title: 'Strava: Track Running, Cycling & Swimming'})
+		})
+	})
+
+	describe('Find app by id tests', () => {
+		beforeEach(async () => {
+			req['query']['userName'] = faker.name.findName()
+			req['query']['password'] = faker.internet.password()
+			await getBearerToken(req, resSpy)
+			headers['authorization'] = resSpy.send.args[0][0]
+		})
+
+		it('Should pass with valid app id', async () => {
+			const response = await axios.get(`${serverUrl}/io.strongapp.strong`, {headers: headers})
+			expect(response).toMatchObject({status: 200})
+			expect(response).toMatchObject({statusText: 'OK'})
+			expect(response.data).toMatchObject({title: 'Strong - Workout Tracker Gym Log'})
```

Comment:
Our endpoint has more data than just the 'title' key, and right now the tests are not covering the actual endpoint response.
So actually there might be a bug where we don't get any other data(beside the title) and we won't be able to catch such a bug.
I would create an object of 'expectedResponse' (in a different file - expected-data.js file), which will includes all the keys the response is actually returning.

Same for here and the search endpoint btw.